app/models/registry.rb

```diff
@@ -0,0 +1,61 @@
+# model that represents a registry
+class Registry < ActiveRecord::Base
+  has_many :registry_mirrors, dependent: :destroy
+  has_one :certificate_service, as: :service, dependent: :destroy
+  has_one :certificate, through: :certificate_service
+
+  validates :name, presence: true, uniqueness: true
+  validates :url, presence: true, uniqueness: true, url: { schemes: ["https", "http"] }
+
+  class << self
+    # create or update suse Registry model
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+    def configure_suse_registry(suse_registry_mirror_params)
+      if suse_registry_mirror_params.blank?
+        cleanup
+        return []
+      end
+
+      errors      = []
+      cert        = suse_registry_mirror_params["certificate"]
+      mirror      = suse_registry_mirror_params["mirror"]
+      name        = suse_registry_mirror_params["name"] || "suse"
+
+      suse_registry_mirror = RegistryMirror.where(name: name).first_or_initialize.tap do |m|
+        m.url = mirror
+        m.registry_id = Registry.find_by(name: "suse").id
+        m.save
+      end
+
+      if suse_registry_mirror.errors.present? || !suse_registry_mirror.persisted?
+        errors << "Registry mirror url #{mirror} doesn't match a registry pattern"
+        return errors
+      end
+
+      if cert.present?
+        certificate = Certificate.find_or_create_by(certificate: cert.strip)
+
+        CertificateService.create(service: suse_registry_mirror, certificate: certificate)
+      elsif suse_registry_mirror.certificate.present?
+        suse_registry_mirror.certificate_service.destroy
+      end
+
+      errors
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+
+    private
+
+    # remove old registries and mirrors from the db if they were deleted in the UI
+    def cleanup
```

Comment:
No need to fetch registries here. We will have only SUSE registry at this point and time and we only need to destroy the associated mirror with it. We can simplify by calling just `RegistryMirror.destroy_all`. Is that right? Or did I miss something?

Reply:
yes for now we dont need to be so intelligent, updated